gpu_bdb/bdb_tools/q28_utils.py

```diff
@@ -267,27 +354,33 @@ def post_etl_processing(client, train_data, test_data):
     y_test = build_labels(test_data)
 
     # Perform ML
-    model = DistMNB(client=client, alpha=0.001)
-    model.fit(X_train, y_train)
+    if isinstance(y_train._meta, cp.ndarray):
+        model = DistMNB(client=client, alpha=0.001)
+        model.fit(X_train, y_train)
+    else:
+        model = ParallelPostFit(estimator=MultNB(alpha=0.001))
+        model.fit(X_train.compute(), y_train.compute())
```

Comment:
You are going to train this on  the client process, is this intentional ?   We should not do anything on the client process.

Reply:
Yep that was a placeholder I meant to replace. Do you have any suggestion for the best way to parallelize SKlearn here? Unfortunately the dask-ml naive bayes model is incompatible with the sparse matrices we use in this query.

Reply:
We should still train on worker processes.  We can probably do something like below.  

```python

est_model = MultNB(alpha=0.001)
X_d = X_train.repartition(npartitions=1).to_delayed()
y_d =  y_train .repartition(npartitions=1).to_delayed()

delayed_model = [delayed(est_model.fit)(x_p, y_p) for x_p, y_p in zip(X_d, y_d)]
model = delayed_model[0].compute()
model = ParallelPostFit(estimator=model)
del est_model
```